ydb/docs/en/core/reference/ydb-cli/interactive-cli.md

```diff
@@ -10,23 +12,88 @@ General format of the command:
+* [Auto completion](#auto-completion)
+* [Special commands](#spec-commands)
+
+## Syntax highlighting {#syntax-highlighting}
+
+![Syntax highlighting](_assets/highlighting.jpg)
+
+Interactive mode supports YQL syntax highlighting, which helps to better understand query structure. Different colors are used for the following groups of elements:
+
+* YQL keywords (SELECT, FROM, WHERE, INSERT, UPDATE and others)
+* Table and column names
+* String literals (text in quotes)
+* Numeric literals
+* Operators (=, <, >, +, - and others)
+* Special characters (brackets, commas, dots)
+* Comments
+
 ## Hotkeys {#hotkeys}
 
+You can use these hotkeys while working in the interactive mode:
+
 | Hotkey        | Description                                                               |
 |---------------|---------------------------------------------------------------------------|
-| `CTRL + D`    | Allows you to exit interactive mode.                                      |
-| `Up arrow`    | Scrolls through query history toward older queries.                       |
-| `Down arrow`  | Scrolls through query history toward newer queries.                       |
-| `TAB`         | Autocompletes the entered text to a suitable YQL command.                 |
-| `CTRL + R`    | Allows searching for a query in history containing a specified substring. |
+| `Up arrow`    | Shows previous query from history.                                        |
+| `Down arrow`  | Shows next query from history.                                            |
+| `TAB`         | Completes the current word based on YQL syntax.                          |
+| `CTRL + R`    | Searches for a query in history containing a specified substring.         |
+| `CTRL + D`    | Exits interactive mode.                                                   |
+
+## Query history {#query-history}
+
+You can navigate through the query history using up and down arrow keys:
+
+![History](_assets/history.gif)
+
+History is stored locally and persists between CLI launches.
+
+A query search function (`CTRL + R`) is also supported:
+
+![Search](_assets/history-search.gif)
+
+## Auto completion {#auto-completion}
+
+Auto completion helps you write queries more efficiently. While typing, it suggests possible completions for the current word based on YQL syntax.
+
+It also searches for schema object names in database where possible.
+
+There are two types of suggestions: auto completion by pressing `TAB` key and interactive hints.
+
+### Auto completion by pressing `TAB` key {#auto-completion-tab}
+
+While in interactive mode, pressing `TAB` key shows a list of suggestions of completion of current word according to the YQL syntax.
+
+![Auto completion](_assets/candidates.gif)
+
+If there is only one available option, pressing TAB will automatically complete the current word.
+If all available options share a common prefix, pressing TAB will automatically insert it.
+
+### Interactive Hints {#interactive-hints}
```

Comment:
The documentation is currently moving from `## Regular case` headers to `## Title Case` headers like this, but at least each file should be consistent in this regard.